packages/app/src/vis-packs/core/hooks.ts

```diff
@@ -100,15 +108,17 @@ export const useCombinedDomain = createMemo(getCombinedDomain);
 const useBaseArray = createMemo(getBaseArray);
 const useApplyMapping = createMemo(applyMapping);
 
-export function useMappedArray<T, U extends T[] | undefined>(
+export function useMappedArray<T, U extends T[] | TypedArray | undefined>(
```

Comment:
I can't use the new `NumArray` type (`TypedArray | number[]`) here, since `useMappedArray` can be called with non-numeric arrays in `MappedMatrixVis`. (Same goes with `getBaseArray` and `applyMapping` since they are called by `useMappedArray`.)

Reply:
I haven't touched `useMappedArrays` yet, which is used in `MappedLineVis` for auxiliary arrays, as I'm getting some typing errors that I need to investigate.